Makefile

```diff
@@ -1,23 +1,43 @@
 
-UNAME := $(shell uname -s)
+GO_BUILD_ENV :=
+GO_BUILD_FLAGS := -tags no_cgo,osusergo,netgo
+MODULE_BINARY := filtered-camera
 
-FLAGS=-tags no_cgo,osusergo,netgo
+ifeq ($(VIAM_TARGET_OS), windows)
+	GO_BUILD_ENV += GOOS=windows GOARCH=amd64
+	MODULE_BINARY = filtered-camera.exe
+endif
 
-ifeq ($(UNAME),Linux)
-    FLAGS += -ldflags="-extldflags=-static -s -w"
+ifeq ($(VIAM_TARGET_OS),linux)
+    GO_BUILD_FLAGS += -ldflags="-extldflags=-static -s -w"
 endif
 
-filtered-camera: Makefile *.go cmd/module/*.go
-	go build $(FLAGS) -o filtered-camera cmd/module/cmd.go
+$(MODULE_BINARY): Makefile *.go cmd/module/*.go
+	$(GO_BUILD_ENV) go build $(GO_BUILD_FLAGS) -o $(MODULE_BINARY) cmd/module/cmd.go
+
+module.tar.gz: meta.json $(MODULE_BINARY)
+	tar czf $@ meta.json $(MODULE_BINARY) 
+	git checkout meta.json
+
+ifeq ($(VIAM_TARGET_OS), windows)
+module.tar.gz: fix-meta-for-win
+else
+module.tar.gz: strip-module
```

Comment:
I'm surprised you can add extra dependencies to a target after defining the target above. but there's all kinds of weird rules about makefile syntax, and if it works, I can't argue.